### ISSUE TITLE
Unity + Parse + iOS NSURLErrorDomain error -1012

### ISSUE THREAD

**Issue by Bestlis** [ParsePlatform/Parse-SDK-dotNET#116]
Hello!
I try to integrate parse into my iOS game.

When I try get info from DB I get exception

Exception: {"error":"The operation couldn’t be completed. (NSURLErrorDomain error -1012.)"}

Here is my code:

var query = ParseObject.GetQuery(ParseTableName).WhereEqualTo("DeviceId", SystemInfo.deviceUniqueIdentifier);
var res = query.FindAsync();

while (!res.IsCompleted)
yield return null;

if (res.IsFaulted)
{
print("Parse exception: " + res.Exception.Message);

foreach (var excep in res.Exception.InnerExceptions)
print("Exception: " + excep.Message);

yield break;
}

Does anybody know what is my problem?
Thanks!

**Comment by hallucinogen** [comment]
Hi there. Thank you for your report.

The last time it happened to me was because I was using a combination of Unity 4.x + IL2CPP + running it on device + accessing HTTPS. Can you try updating your Unity into 5.x, preferably 5.2 and try to compile it without IL2CPP backend?

Thanks!

**Comment by Bestlis** [comment]
Hallucinogen:
I use Unity 5.2.3p2

**Comment by hallucinogen** [comment]
Are you compiling using IL2CPP? What device are you using to test the game? Can you execute a simple save?

```
var obj = new ParseObject("SomeClass");
obj.SaveAsync();
```

**Comment by Bestlis** [comment]
I think I find out solution: I didn't create class in Parse console.
When I created, then error dissapear.

**Comment by Bestlis** [comment]
But now I have another problem: no data save to Parse. Only ObjectId, createdAt, updatedAt and ACL.
Any ideas?

**Comment by hallucinogen** [comment]
Have you saved the data properly? Have the object that you said only have ObjectId, CreatedAt, UpdatedAt and ACL been saved properly? How did you verify that the object only contain those data? Try the following code:

```
var obj = new ParseObject("SomeClass");
obj["foo"] = "bar";
obj.SaveAsync().ContinueWith(_ => {
  var query = ....; // Do the query here
  return query.FindAsync();
}).Unwrap().ContinueWith(result => {
  // Print result.result here.
});
```

And FYI, if you save a field called `fieldName` in a `ParseObject`, you can access it by calling `obj["fieldName"]` - not `obj.fieldName`. This is stated in Parse documentation: https://parse.com/docs/unity/guide#objects-retrieving-objects

**Comment by Bestlis** [comment]
Hello, hallucinogen.
Thanks for reply!

This works fine:

var obj = new ParseObject ("SolitaireWithFriendsiOS");
   obj ["foo"] = "bar";
    obj.SaveAsync ().ContinueWith (t =>
{
    print ("Result: " + t.IsCompleted);
});

This throws exception NSURLErrorDomain error -1012: 

IEnumerator ParseSave()
{
	var obj = new ParseObject ("SolitaireWithFriendsiOS");
	obj ["foo"] = "bar";
	var query = obj.SaveAsync ();
		            
	while (!query.IsCompleted)
		yield return null;

	if (query.IsFaulted) {
		print ("Parse exception: " + query.Exception.Message);
				
		foreach (var excep in query.Exception.InnerExceptions)
			print ("Exception: " + excep.Message);
	} else
		print ("RESULT: " + query.IsCompleted);
}

**Comment by hallucinogen** [comment]
Not sure about this. Both essentially do the same thing (though your code is not actually doing query, but only saving).

`NSURLErrorDomain error -1012` happens usually if there're internal WWW error from Unity. In my previous case it was because Unity couldn't handle HTTPS, but it was fixed in 5.2. I suggest to look around on the usage of Unity WWW using co-routine/generator function since it seems the code fails while executing a network request inside a co-routine.
